examples/tutorials/matmul/mxfp_matmul.cpp

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+* Copyright 2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/// @example mxfp_matmul.cpp
+/// > Annotated version: @ref mxfp_matmul_cpp
+///
+/// @page mxfp_matmul_cpp MatMul Tutorial: MXFP8 Inference
+/// C++ API example demonstrating how one can use
+/// [MatMul](@ref dev_guide_matmul) with MXFP8 datatype in inference.
+///
+/// Concepts:
+/// - Dynamic quantization compliant with MX specification
+///   - Scales: dnnl::primitive_attr::set_scales()
+/// - Create primitive once, use multiple times
+///
+/// @include mxfp_matmul.cpp
+
+#include <cassert>
+#include <cctype>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <random>
+#include <stdexcept>
+#include <vector>
+
+#include "oneapi/dnnl/dnnl.hpp"
+
+#include "example_utils.hpp"
+
+using namespace dnnl;
+
+namespace {
+
+void init_vector(std::vector<float> &v) {
+    std::mt19937 gen;
+    std::uniform_real_distribution<float> u(0, 1);
+    for (auto &e : v)
+        e = u(gen);
+}
+
+uint8_t f32_to_e8m0(const float &a) {
+    // Note: memcpy can be replaced with bit_cast in C++20
+    uint32_t a_s32;
+    std::memcpy(&a_s32, &a, sizeof(float));
+    uint8_t a_e8m0 = (a_s32 >> 23) & 0xff;
+    return a_e8m0;
+}
+
+float e8m0_to_f32(const uint8_t a) {
+    // Note: memcpy can be replaced with bit_cast in C++20
+    float r_f32;
+    uint32_t r_s32 = uint32_t(a) << 23;
+    std::memcpy(&r_f32, &r_s32, sizeof(float));
+    return r_f32;
```

Comment:
The spec says e8m0 has neither inf nor zero:
```suggestion
    if (a == 0xff) return std::numeric_limits<float>::quiet_NaN();
    return std::pow(2.0f, int(a) - 127);
```